src/sidebar/components/test/thread-list-omega-test.js

```diff
@@ -0,0 +1,280 @@
+    return mount(
+      <ThreadList
+        thread={fakeTopThread}
+        $rootScope={fakeRootScope}
+        {...props}
+      />,
+      { attachTo: fakeScrollContainer }
+    );
+  }
+
+  beforeEach(() => {
+    fakeDebounce = sinon.stub().returns(() => null);
+    fakeDomUtil = {
+      getElementHeightWithMargins: sinon.stub().returns(0),
+    };
+    fakeMetadata = {
+      isHighlight: sinon.stub().returns(false),
+      isReply: sinon.stub().returns(false),
+    };
+
+    fakeRootScope = {
+      eventCallbacks: {},
+
+      $apply: function (callback) {
+        callback();
+      },
+
+      $on: function (event, callback) {
+        if (event === events.BEFORE_ANNOTATION_CREATED) {
+          this.eventCallbacks[event] = callback;
+        }
+      },
+
+      $broadcast: sinon.stub(),
+    };
+
+    fakeScrollContainer = document.createElement('div');
+    fakeScrollContainer.className = 'js-thread-list-scroll-root';
+    fakeScrollContainer.style.height = '2000px';
+    document.body.appendChild(fakeScrollContainer);
+
+    fakeStore = {
+      clearSelection: sinon.stub(),
+    };
+
+    fakeTopThread = {
+      id: 't0',
+      annotation: { $tag: 'myTag0' },
+      children: [
+        { id: 't1', children: [], annotation: { $tag: 't1' } },
+        { id: 't2', children: [], annotation: { $tag: 't2' } },
+        { id: 't3', children: [], annotation: { $tag: 't3' } },
+        { id: 't4', children: [], annotation: { $tag: 't4' } },
+      ],
+    };
+
+    fakeVisibleThreadsUtil = {
+      calculateVisibleThreads: sinon.stub().returns({
+        visibleThreads: fakeTopThread.children,
+        offscreenUpperHeight: 400,
+        offscreenLowerHeight: 600,
+      }),
+      THREAD_DIMENSION_DEFAULTS: {
+        defaultHeight: 200,
+      },
+    };
+
+    $imports.$mock(mockImportedComponents());
+    $imports.$mock({
+      'lodash.debounce': fakeDebounce,
+      '../store/use-store': callback => callback(fakeStore),
+      '../util/annotation-metadata': fakeMetadata,
+      '../util/dom': fakeDomUtil,
+      '../util/visible-threads': fakeVisibleThreadsUtil,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('works', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(wrapper.find('section').exists());
+  });
+
+  it('calculates visible threads', () => {
+    createComponent();
+
+    assert.calledWith(
+      fakeVisibleThreadsUtil.calculateVisibleThreads,
+      fakeTopThread.children,
+      sinon.match({}),
+      0,
+      sinon.match.number
+    );
+  });
+
+  context('new annotation created in application', () => {
+    it('attaches a listener for the BEFORE_ANNOTATION_CREATED event', () => {
+      fakeRootScope.$on = sinon.stub();
+
+      createComponent();
+
+      assert.calledWith(
+        fakeRootScope.$on,
+        events.BEFORE_ANNOTATION_CREATED,
+        sinon.match.func
+      );
+    });
+
+    it('clears the current selection in the store', () => {
```

Comment:
There is no way AFAICT that I can introspect into component state and verify that the `scrollToId` state was updated in component https://github.com/hypothesis/client/pull/2044/files#diff-0aa90e2743ab4cd906abb9caa7f3364bR68 , thus these tests check to see if`store.clearSelection` was called, as that can be mocked and checked.

Reply:
> There is no way AFAICT that I can introspect into component state and verify that the scrollToId state was updated

Indeed. The fact this state even exists is an implementation detail - you might want to do it differently in future. The externally visible behavior we're interested in is that the element identified by `getScrollContainer()` has its `scrollTop` adjusted - in other words, the thread is scrolled into view. Looks like you've got a separate test for that.